components/scream/src/diagnostics/potential_temperature.hpp

```diff
@@ -0,0 +1,94 @@
+#ifndef EAMXX_POTENTIAL_TEMP_DIAGNOSTIC_HPP
+#define EAMXX_POTENTIAL_TEMP_DIAGNOSTIC_HPP
+
+#include "share/atm_process/atmosphere_diagnostic.hpp"
+#include "share/util/scream_common_physics_functions.hpp"
+
+namespace scream
+{
+
+/*
+ * This diagnostic will produce the potential temperature.
+ */
+
+class PotentialTemperatureDiagnostic : public AtmosphereDiagnostic
+{
+public:
+  template <typename S>
+  using SmallPack     = ekat::Pack<S,SCREAM_SMALL_PACK_SIZE>;
+
+  using Spack         = SmallPack<Real>;
+  using Pack          = ekat::Pack<Real,Spack::n>;
+  using PF            = scream::PhysicsFunctions<DefaultDevice>;
+  using KT            = KokkosTypes<DefaultDevice>;
+  using view_2d       = typename KT::template view_2d<Spack>;
+  using view_2d_const = typename KT::template view_2d<const Spack>;
+
+  // Constructors
+  PotentialTemperatureDiagnostic (const ekat::Comm& comm, const ekat::ParameterList& params);
+
+  // Set type to diagnostic
+  AtmosphereProcessType type () const { return AtmosphereProcessType::Diagnostic; }
+
+  // The name of the diagnostic
+  std::string name () const { return "Potential Temperature"; } 
+
+  // Get the required grid for the diagnostic
+  std::set<std::string> get_required_grids () const {
+    static std::set<std::string> s;
+    s.insert(m_params.get<std::string>("Grid"));
+    return s;
+  }
+
+  // Set the grid
+  void set_grids (const std::shared_ptr<const GridsManager> grids_manager);
+
+  // Actual diagnostic calculation 
+  struct run_diagnostic_impl {
+    run_diagnostic_impl() = default;
+    // Functor for Kokkos loop
+    KOKKOS_INLINE_FUNCTION
+    void operator()(const int icol) const {
+      for (int jpack=0;jpack<m_npack;jpack++) {
```

Comment:
This for loop wont be performant on GPU, since access is not coalesced. One possibility is to incorporate `m_npacks` in the range policy size, remove this for loop, and retrieve icol/ipack as
```
void operator() (const int& idx) const {
   const int icol    = idx / m_ncol;
   const int jpack = idx % m_ncol;
   ...
}
```

Reply:
okay cool.  I copy/pasted this from elsewhere.  Should I clean up those other CMakeLists.txt files?